iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -210,16 +238,15 @@ private async Task SendMessagesAsync(CancellationToken cancellationToken)
                         }
                         catch (IotHubException ex) when (ex.IsTransient)
                         {
-                            // Inspect the exception to figure out if operation should be retried, or if user-input is required.
-                            _logger.LogError($"An IotHubException was caught, but will try to recover and retry: {ex}");
+                            _logger.LogError($"A transient IotHubException was caught, will retry sending the message: {ex}");
                         }
-                        catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
-                        {
-                            _logger.LogError($"A network related exception was caught, but will try to recover and retry: {ex}");
+                        catch (UnauthorizedException ex)
+                        {
+                            _logger.LogError($"An UnauthorizedException was caught, recovery will be handled by the ConnectionStatusChangeHandler: {ex}");
                         }
-                        catch (Exception ex)
+                        catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
                         {
-                            _logger.LogError($"Unexpected error {ex}");
```

Comment:
no longer consuming a non-recoverable exception, instead causing the application to terminate